Middleware/NewRelicMiddleware.php

```diff
@@ -28,8 +29,15 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
         if (!$this->newrelicManager->isEnabled()) {
             return $stack->next()->handle($envelope, $stack);
         }
+
         $this->newrelicManager->startTransaction();
-        $this->newrelicManager->nameTransaction(get_class($envelope->getMessage()));
+        $message = $envelope->getMessage();
+        if ($message instanceof NameableInterface) {
```

Comment:
Can we isolate this logic in a separate class?
This way this will be easier to test, and we can use early returns (we are not using else in this repository).

Can you also include the test for this?

Reply:
done :-)

Reply:
done :-)